model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/ReplaceFunction.java

```diff
@@ -0,0 +1,35 @@
+package org.keycloak.models.sessions.infinispan.changes;
+
+import org.infinispan.util.function.SerializableBiFunction;
+import org.keycloak.models.sessions.infinispan.entities.SessionEntity;
+
+import java.io.Serial;
+import java.util.UUID;
+
+/**
+ * Performs an entity replacement in Infinispan, using its versions instead of equality.
+ *
+ * @param <K> The Infinispan key type.
+ * @param <T> The Infinispan value type (Keycloak entity)
+ */
+public class ReplaceFunction<K, T extends SessionEntity> implements SerializableBiFunction<K, SessionEntityWrapper<T>, SessionEntityWrapper<T>> {
+
+
+    @Serial
+    private static final long serialVersionUID = -3317222251828996255L;
+    private final UUID expectedVersion;
+    private final SessionEntityWrapper<T> newValue;
```

Comment:
JBoss-Marshalling uses reflection to fetch the fields of the object instance and serializes each one. When it reaches the `newValue` field, it goes through the JBoss-Marshaller and finds the `Externalizer`, using it.

Reply:
Thank you for this answer, this resolves it for me.

Reply:
Thinking about it a bit more, I now understand that this is always serialised using the JBoss marshaller. Everywhere we use the marshaller today, we use an externalizer which as the first value sends a version number.

Could you please an externalizer for this class? I know this class is unlikely to change until, still I want to be on the safe side here.

We also talked about backporting and mentioned it will be a breaking change. with the discussion we had in the F2F, could we backport the new class and it's externalizer in one patch release, and then change to compute in the next patch release? As long as people move only by a single patch release this would then be non-breaking. WDYT? IMHO if we want eventually guarantee it, we could as well start practicing it.

Reply:
Sure, I've added an `AdvancedExternalizer` instead. I didn't add an externalizer for WildFly because Keycloak is no longer based in WF. 

If you mean RHBK patch releases, yes we can do that and it should work.